### PR TITLE
Create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by jfcg**
Add Github CodeQL security checks for Go.

### CODE REVIEW

**Comment by wonderflow** [comment]
> CodeQL / Analyze (go) (pull_request) Successful in 26m

As you see, the code analyze took too much time, we didn't earn enough to add it into our CI and I didn't see any valuable results it reported.

**Comment by jfcg** [comment]
CodeQL checks are especially valuable for web apps, but it is up to the project of course.

**Comment by jfcg** [comment]
Btw in my fork under `/security/code-scanning` I see:
kubevela

**Comment by wonderflow** [comment]
@jfcg Where can I see these warnings?

**Comment by jfcg** [comment]
Here https://github.com/oam-dev/kubevela/pull/1470/checks?check_run_id=2327483395

**Comment by wonderflow** [comment]
@jfcg Thanks, I think this report is valuable, but it's cost too much for running it for every pull request. Do you have any good suggestion? Or add it as a periodically job in action?

**Comment by jfcg** [comment]
maybe just after every push on release-1.0 ? I can commit that to this MR.

**Comment by wonderflow** [comment]
> maybe just after every push on release-1.0 ? I can commit that to this MR.

Good idea, I think every push to `release-*` and `master` should be good

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master, release-1.0, release-0.3 ]
```
Comment (wonderflow) [inline .github/workflows/codeql-analysis.yml:5]:
```suggestion
    branches: [ master, release-* ]
```

**Review by wonderflow — APPROVED** [review]
LGTM thanks

**Pull request merged**